vital/bindings/python/vital/types/image_container.cxx

```diff
@@ -151,33 +132,23 @@ PYBIND11_MODULE(image_container, m)
         >>> assert np.all(np_img == np_img2)
     )")
 
-  .def(py::init(&new_cont), py::arg("image"))
-
-  .def_static("fromarray",
-    [](py::array& arr) -> s_image_cont {
-      auto locals = py::dict(py::arg("arr")=arr);
-      py::exec(R"(
-          from vital.types import ImageContainer
-          from PIL import Image as PILImage
-          from vital.util import VitalPIL
-          pil_img = PILImage.fromarray(arr)
-          vital_img = VitalPIL.from_pil(pil_img)
-          self = ImageContainer(vital_img)
-          )", py::globals(), locals);
-      return locals["self"].cast<s_image_cont>();
-    }, py::doc(R"(
-    Create an ImageContainer from a numpy array
-
-    Example:
-        >>> from vital.types import ImageContainer
-        >>> import numpy as np
-        >>> np_img = (np.random.rand(10, 20, 3) * 255).astype(np.uint8)
-        >>> self = ImageContainer.fromarray(np_img)
-        >>> print(str(self))
-        <ImageContainer(whd=20x10x3)>
-        >>> np_img2 = self.asarray()
-        >>> assert np.all(np_img == np_img2)
-    )")
-  )
-  ;
+  .def(py::init(&kwiver::vital::python::image_container::new_cont), py::arg("image"))
+
+  // Create initialzer based on numpy array type
+  #define fromarray( T ) \
+  .def_static("fromarray", \
+              &kwiver::vital::python::image_container::new_image_container_from_numpy<T>, \
+              py::arg("array"),\
+      py::doc("Create an ImageContainer from a numpy array"))
+  fromarray( uint8_t )
```

Comment:
Unless this is following some C++ convention, it may be more clear to rename this macro from `fromarray` to `def_fromarray` to indicate that the macro is creating a definition.